kroxylicious-filter/src/test/java/io/strimzi/kafka/topicenc/kroxylicious/KafkaAssertions.java

```diff
@@ -0,0 +1,45 @@
+package io.strimzi.kafka.topicenc.kroxylicious;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class KafkaAssertions {
+
+    public static void assertOnlyRecordEquals(Consumer<String, byte[]> kafkaClusterConsumer, String topic, byte[] expected) {
```

Comment:
```suggestion
    public static void assertSingletonRecordEquals(Consumer<String, byte[]> kafkaClusterConsumer, String topic, byte[] expected) {
```

With the knock on implications for `onlyRecord` -> `singletonRecord` in general.

Reply:
What's the distinction? Is only a bit vague?

Reply:
In assertj only is duplicate insensitive so singleton might give a better description.

Reply:
Yeah, `only` didn't mean much and could be read as `assertOnlyOneRecordMatches` rather than what we intend that there is a single message read.